specs/DesignDiscussions/InputValidation.md

```diff
@@ -0,0 +1,260 @@
+
+> Note on this card: This "Wizard" scenario is implemented here via toggle visibility. We should consider whether we want to support a first class wizard experience where a new element type could list a series of subcards to display in order. Whether or not we ever introduce such a concept, the point remains that toggle logic may be complex, and we shouldn't be arbitrarily toggling things visible outside the scope of what the card author has defined in their toggle buttons.
+
+#### Current Card/Container
+
+Considering the food order card above, one option in show card scenarios may be to validate the current show card, or the show card and any parent cards. This works well for the food order card, but doesn't help us with card scenarios that include toggles. It may also not be the correct behavior for a mid card action set containing a show card. While it would be nice if we could provide an algorithm that deduces which inputs should be validated, it would be tricky to create an algorithm that gracefully handles all cases. Additionally, the more complicated the algorithm, the harder it is for card authors to understand the expected behavior.
+
+#### Card Author Control
+
+As the discussion above shows, depending on the card scenario, it may be difficult for us to determine which fields the card author wants to validate. Any of the above approaches may be correct for *some* cards, but none are appropriate for all cards. In order to address this, the best approach is to allow the card author to specify which inputs are associated with a given action. We will implement this via a new `inputs` property on Action types. The inputs listed will be the ones that are validated, and also the ones that are returned on Submit.
+
+The first option available for inputs is "All." This will validate and submit all inputs on the card. This is the current behavior, and would be the default for submit actions.
+```json
+{
+	"type": "Action.Submit",
+	"title": "Submit",
+	"inputs": "All",
+}
+```
+
+The second option would be "None". This would be useful for, for example, a "Cancel" button where validation is not required, and would be the default for non-Submit (or Invoke action being discussed for Universal Actions) action types (see further discussion of validation for different action types below).
+```json
+{
+	"type": "Action.Submit",
+	"title": "Cancel",
+	"inputs": "None",
+}
+```
+For authors that need more granular control of which inputs to validate, we would also allow them to specify a list of inputs, such as the following:
+```json
+{
+	"type": "Action.Submit",
+	"title": "Cancel",
+	"inputs": ["input1Id", "input2Id"] 
+}
+```
+In order to allow authors to group their inputs, we could also accept container ids. This allows the card author to functionally define a "form" and indicate that the submit button relates to that form:
+```json
+{
+	"type": "Action.Submit",
+	"title": "Cancel",
+	"inputs": ["input1Id", "input2Id", "formContainerId"] 
```

Comment:
Discussion: Can this be a card Id? We don't have card ids in shared model today, but we do in JavaScript. We probably should have them in all platforms, at which point yes, this could be a card id.

Reply:
Updated the spec to call out the schema change for card ids.

---
In reply to: [367156982](https://github.com/microsoft/AdaptiveCards/pull/3663#discussion_r367156982) [](ancestors = 367156982)